astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -304,16 +304,22 @@ def _get_components_and_classes(self):
 
         if self.has_celestial:
 
-            frame = wcs_to_celestial_frame(self)
+            try:
+                frame = wcs_to_celestial_frame(self)
+            except ValueError:
+                # Some WCSes, e.g. solar, can be recognized by WCSLIB as being
```

Comment:
You could get real froody here and attempt to import `sunpy.coordinates` :stuck_out_tongue: